caas/kubernetes/provider/k8s.go

```diff
@@ -1601,14 +1663,38 @@ func (k *kubernetesClient) ensureStatefulSet(spec *apps.StatefulSet, existingPod
 	existing.Spec.Template.Spec.ServiceAccountName = existingPodSpec.ServiceAccountName
 	existing.Spec.Template.Spec.AutomountServiceAccountToken = existingPodSpec.AutomountServiceAccountToken
 	// NB: we can't update the Spec.ServiceName as it is immutable.
-	_, err = api.Update(existing)
+	_, err = k.updateStatefulSet(existing)
 	return errors.Trace(err)
 }
 
-// createStatefulSet deletes a statefulset resource.
-func (k *kubernetesClient) createStatefulSet(spec *apps.StatefulSet) error {
-	_, err := k.client().AppsV1().StatefulSets(k.namespace).Create(spec)
-	return errors.Trace(err)
+func (k *kubernetesClient) createStatefulSet(spec *apps.StatefulSet) (*apps.StatefulSet, error) {
```

Comment:
as a driveby can we separate these out to a new statefulset.go file like was done for daemonset
(plus the delete, list etc)